### ISSUE TITLE
heading-has-content false positive for object spread

### ISSUE THREAD

**Issue by wozitto** [jsx-eslint/eslint-plugin-jsx-a11y#980]
The code throws an error.
```jsx
const SectionTitle = (props) => (
  <h2 {...props} />
);
```

However, this code does not throw an error.
```jsx
const SectionTitle = (props) => (
  <h2 {...props} >{props.children}</h2>
);
```

Both syntaxes produce the same output when rendered, which leads me to believe that the first syntax should not result in an error either.

Version: 6.8.0
<img width="746" alt="スクリーンショット 2024-04-17 22 40 42" src="https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/assets/74745265/17ce2972-7246-4a44-99bf-075d871f26c1">

**Comment by ljharb** [comment]
This is the natural limits of syntactic analysis - we can’t know if children is included in the props, so we have to require a syntactic children placement.
